Feat/1010 signup

zjimf opened the pull request:
Hi guys,
I finish the ui, but only static.
please go into the URL below and check, tks

localhost:3000/auth/verifyEmail
localhost:3000/auth/signUp

zjimf committed:
style: uniform space interval

zjimf committed:
feat: complete verifyEmail static ui
no logic has been added yet

zjimf committed:
refactor: remove parameter

zjimf committed:
feat: add MUI and MUI icon library

zjimf committed:
style: uniform height

zjimf committed:
feat: complete signUp static ui

zjimf committed:
fix: remove interface temporary

HenryHung90 committed:
doc: add npm uji

1092024 commented:
- 字型不一樣，字型應設定為Inter

- 缺少input裡的placeholder
https://www.figma.com/design/CDqCjF4aRruKELFRNMcENz/Stamind?node-id=167-5471&t=ubvmejmNBTaX0Sds-4

- input在focus狀態，input的邊框有雙層(白色＋藍色邊框)，應修正為只有單層(藍色邊框)
<img width="481" alt="截圖 2024-10-27 晚上11 20 47" src="https://github.com/user-attachments/assets/0c228e5a-ca65-48dc-bbaa-b2ad4af0b758">

- button 在不能點擊的狀態，設計搞上為藍色(但與可點擊狀態的藍色不同)
<img width="175" alt="截圖 2024-10-27 晚上11 25 41" src="https://github.com/user-attachments/assets/54513fc2-599c-446a-9e8d-ccc3eb671bb5">

zjimf committed:
refactor: split logic into multiple files

- components: split into Inputfield(for username) and PasswordField(for
  password and confirm password)
- hooks: unified processing of status updates, verification conditions, display control of input boxes, etc.
- utils: only perform data and logic processing to ensure centralized and unified verification logic

zjimf committed:
fear: add google btn

zjimf committed:
feat: add placeholder

zjimf committed:
fix: wrong name in code and file name

zjimf committed:
feat: add email field

zjimf committed:
fix: adding variable in error state

zjimf commented:
@1092024 I'll fix the error tomorrow, tks

zjimf committed:
fix: focus border color error

zjimf committed:
fix: empty email message show error

zjimf committed:
fix: button disable opasity wrong

zjimf committed:
feat: font-family change to Inter

zjimf commented:
@1092024 @HenryHung90 check this plz, tks 😄

HenryHung90 commented:
@zjimf  PM must approve first, then I go further ~

1092024 commented:
@zjimf 
在所有input錯誤都顯示的狀態，在我的畫面無全螢幕下(macbook 13寸)，會有input擠到上面標題，輸入資料區應該往下擴張
謝謝！
<img width="1470" alt="截圖 2024-10-28 下午2 26 46" src="https://github.com/user-attachments/assets/4b3ad5a7-bb38-484d-84be-4d03374b7389">

zjimf committed:
fix: extrude the layout to the title

HenryHung90 committed:
fix: change Email, input field height, change passwordField props and add custom class

zjimf committed:
style: uniform the gap

zjimf committed:
Merge branch 'main' into feat/1010_signup

zjimf committed:
refactor: redirect to component

HenryHung90 reviewed (approved):
Good to go
